core/src/main/java/com/datastrato/gravitino/utils/PrincipalUtils.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2023 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.utils;
+
+import com.datastrato.gravitino.UserPrincipal;
+import com.datastrato.gravitino.auth.AuthConstants;
+import java.security.Principal;
+import java.security.PrivilegedActionException;
+import java.security.PrivilegedExceptionAction;
+import javax.security.auth.Subject;
+
+@SuppressWarnings("removal")
+public class PrincipalUtils {
```

Comment:
I extract the doAs and getCurrentUser from Hadoop UserInformation.

Reply:
This api will be removed in the future. But JDK doesn't provide alternative solution. Hadoop will encounter the similar situation.